src/Injector.php

```diff
@@ -69,7 +69,9 @@ public function withCacheReflections(bool $cacheReflections = true): self
      * This will pass the string `'Hello World!'` as the first argument, and a formatter instance created
      * by the DI container as the second argument.
      *
-     * @param callable $callable callable to be invoked.
+     * @template TReturn
+     *
+     * @param callable(): TReturn $callable callable to be invoked.
```

Comment:
I don't think it's a good idea. We are going to render phpdoc with phpdocumentor and alike tools for API docs, and it does not support Psalm annotations as far as I know. Same for non-PhpStorm editors/IDEs.